sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -75,6 +75,83 @@ import org.apache.spark.sql.types.IntegerType
  *       LocalTableScan [...]
  * }}}
  *
+ * Second example: aggregate function without distinct and with filter clauses (in sql):
+ * {{{
+ *   select count(distinct cat1) as cat1_cnt, count(distinct cat2) as cat2_cnt,
+ *     sum(value) filter (where id > 1) as total
+ *   from data group by key
```

Comment:
nit: can we make the SQL more readable?
```
SELECT
  COUNT ...
  COUNT ...
  SUM ...
FROM ...
GRUP BY ...
```
e.g. upper case the keyword and better indentation

Reply:
OK